platform/mellanox/mlnx-platform-api/sonic_platform/modules_mgmt.py

```diff
@@ -469,47 +492,40 @@ def check_module_type(self, port, module_sm_obj, dynamic=False):
-                module_fd_freq_support_path = SYSFS_INDEPENDENT_FD_FREQ_SUPPORT.format(port)
-                val_int = utils.read_int_from_file(module_fd_freq_support_path)
-                if 1 == val_int:
-                    # read the module maximum supported clock of Management Comm Interface (MCI) from module EEPROM.
-                    # from byte 2 bits 3-2:
-                    # 00b means module supports up to 400KHz
-                    # 01b means module supports up to 1MHz
-                    logger.log_info(f"check_module_type reading mci max frequency for port {port}")
-                    read_mci = xcvr_api.xcvr_eeprom.read_raw(2, 1)
-                    logger.log_info(f"check_module_type read mci max frequency {read_mci} for port {port}")
-                    mci_bits = read_mci & 0b00001100
-                    logger.log_info(f"check_module_type read mci max frequency bits {mci_bits} for port {port}")
-                    # Then, set it to frequency Sysfs using:
-                    # echo <val> > /sys/module/sx_core/$asic/$module/frequency //  val: 0 - up to 400KHz, 1 - up to 1MHz
-                    indep_fd_freq = SYSFS_INDEPENDENT_FD_FREQ.format(port)
-                    utils.write_file(indep_fd_freq, mci_bits)
+            self.update_frequency(port, xcvr_api)
+            logger.log_info("check_module_type port {} setting STATE_SW_CONTROL module ID {} due to flat_mem device".format(xcvr_api, port))
+            return STATE_SW_CONTROL
+        else:
+            # QSFP-DD, OSFP, QSFP+C, QSFP+, QSFP28 - only these 5 active form factors are supported currently as independent module - SW controlled
+            if self.is_supported_for_software_control(xcvr_api):
+                power_cap = self.check_power_cap(port, module_sm_obj)
+                if power_cap is STATE_POWER_LIMIT_ERROR:
+                    module_sm_obj.set_final_state(STATE_POWER_LIMIT_ERROR)
+                    return STATE_POWER_LIMIT_ERROR
+                self.update_frequency(port, xcvr_api)
+                logger.log_info("check_module_type port {} setting STATE_SW_CONTROL module ID {} due to supported paged_mem device".format(xcvr_api, port))
                 return STATE_SW_CONTROL
+            else:
+                return STATE_FW_CONTROL
 
     def check_power_cap(self, port, module_sm_obj, dynamic=False):
         logger.log_info("enter check_power_cap port {} module_sm_obj {}".format(port, module_sm_obj))
         sfp = sfp_module.SFP(port)
         xcvr_api = sfp.get_xcvr_api()
-        field = xcvr_api.xcvr_eeprom.mem_map.get_field(consts.MAX_POWER_FIELD)
+        if isinstance(xcvr_api, cmis.CmisApi):
+            field = xcvr_api.xcvr_eeprom.mem_map.get_field(consts.MAX_POWER_FIELD)
+        elif isinstance(xcvr_api, sff8636.Sff8636Api) or isinstance(xcvr_api, sff8436.Sff8436Api):
+            field = xcvr_api.xcvr_eeprom.mem_map.get_field(consts.POWER_CLASS_FIELD)
```

Comment:
From SDK API doc I can see that power_limit only supports CMIS now, could you please double confirm with SDK team to make sure that it will be implemented for SFF?

Reply:
I synced with Andy for the second time regarding the way to get the max power of an SFF module and this time I got a more detailed answer. Code was updated.

Reply:
According to SDK API, the sysfs power_limit unit is 0.25W. But what we got from EEPROM for SFF8636/8436 is based no unit 1W. Could you please check if we need handle this? Maybe we should check with Tamar about the power_limit implementation of SFF8436/8636.

Reply:
Can you show me where you saw this 0.25W unit in the SDK API? and how do you know that the units we get from the EEPROM for SFF8636/8436 is based on 1W?

Reply:
Answered in teams

Reply:
Checked with Tamar, she thinks we indeed need to multiply the sysfs value by 4 before comparing it to the value from the EEPROM. Code was updated accordingly.